test/distributed/org/apache/cassandra/distributed/test/SchemaTest.java

```diff
@@ -95,21 +98,37 @@ private void selectSilent(Cluster cluster, String name)
     @Test
     public void schemaReset() throws Throwable
     {
+        CassandraRelevantProperties.MIGRATION_DELAY.setLong(10000);
+        CassandraRelevantProperties.SCHEMA_PULL_INTERVAL.setLong(10000);
         try (Cluster cluster = init(Cluster.build(2).withConfig(cfg -> cfg.with(Feature.GOSSIP, Feature.NETWORK)).start()))
         {
             cluster.schemaChange("CREATE TABLE " + KEYSPACE + ".tbl (pk INT PRIMARY KEY, v TEXT)");
 
             assertTrue(cluster.get(1).callOnInstance(() -> Schema.instance.getTableMetadata(KEYSPACE, "tbl") != null));
             assertTrue(cluster.get(2).callOnInstance(() -> Schema.instance.getTableMetadata(KEYSPACE, "tbl") != null));
 
+            // now we have a table tbl in the schema of both nodes
+
             cluster.get(2).shutdown().get();
 
-            // when schema is removed and there is no other node to fetch it from, node 1 should be left with clean schema
-            cluster.get(1).runOnInstance(() -> Schema.instance.resetLocalSchema());
+            // when there is no node to fetch the schema from, reset local schema should immediately fail
```

Comment:
AFAIU it worked in the way that if we don't get a schema from some other node, we will be left in an inconsistent state, where we have `Keyspace` instances for keyspaces which are missing in the local schema. We would have to restart the node in such case. Also if some keyspaces were modified on the other node in the meantime, say a table was dropped - that table will not be dropped on the node were we called reset because from that node's view point, nothing has been dropped, and the stale data files will be left.